frontend/src/components/AmountInput.tsx

```diff
@@ -0,0 +1,99 @@
+import { satsToUsd, usdToSats } from "@util/conversions";
+import prettyPrintAmount from "@util/prettyPrintAmount";
+import { usePriceQuery } from "@util/queries";
+import { Dispatch, SetStateAction, useContext, useState } from "react";
+import { inputStyle, selectStyle } from "../styles";
+import { NodeManagerContext } from "./GlobalStateProvider";
+
+type Props = {
+    amountSats: string,
+    setAmount: Dispatch<SetStateAction<string>>,
+    accent: "green" | "blue" | "red",
+    placeholder: string,
+}
+
+export enum Currency {
+    USD = "USD",
+    SATS = "SATS"
+}
+
+export default function AmountInput({ amountSats, setAmount, accent, placeholder }: Props) {
+    const [currency, setCurrency] = useState(Currency.SATS)
+
+    // amountSats will be our source of truth, this is just for showing to the user
+    const [localDisplayAmount, setLocalDisplayAmount] = useState(amountSats)
+
+    const { nodeManager } = useContext(NodeManagerContext);
+
+    const { data: price } = usePriceQuery(nodeManager);
+
+    function setAmountFormatted(value: string) {
+        if (value.length === 0 || value === "0") {
+            setAmount('')
+            setLocalDisplayAmount('')
+            return
+        }
+        //Use a regex to replace all commas and underscores with empty string
+        const amount = value.replace(/[_,]/g, "");
+        let parsedAmount = parseInt(amount)
+        if (typeof parsedAmount === "number" && parsedAmount !== 0 && price) {
+            // If the currency is set to usd, we need to convert the input amount to sats and store that in setAmount
+            if (currency === Currency.USD) {
+                setAmount(usdToSats(parsedAmount, price))
```

Comment:
should this be?
```
setAmount(satsToUSD(parsedAmount, price))
```
This is for where the input is formatting for USD when the Currency.USD option is selected correct?

Reply:
If this is correct I think this is what is causing the input to not be able to have a pre-formatted value of less than $1

Reply:
yeah it's correct, the "source of truth" is sats so we always need to set that.

yeah I gave up before I could manage to support less than $1. maybe a good follow-on pr?